source/sas.cpp

```diff
@@ -382,11 +384,33 @@ bool SAS::Connection::connect_init()
   if (rc != 0)
   {
     SAS_LOG_ERROR("Failed to connect to SAS %s:%s : %d %s", _sas_address.c_str(), SAS_PORT, errno, ::strerror(errno));
-    return false;
+    return -1;
   }
 
   freeaddrinfo(addrs);
 
+  return _sock;
+}
+
+
+bool SAS::Connection::connect_init()
+{
+  int _sock;
+
+  if (_socket_factory)
+  {
+    _sock = _socket_factory(_sas_address.c_str(), SAS_PORT);
+  }
+  else
+  {
+    _sock = get_local_sock();
```

Comment:
I think it would be better if `_socket_factory` and `get_local_socket` took the same arguments.